src/core/thread/mle_router.hpp

```diff
@@ -566,6 +569,7 @@ class MleRouter: public Mle
     uint8_t mMaxChildrenAllowed;
     Child mChildren[kMaxChildren];
 
+    bool    mChallengeValid;
```

Comment:
We could avoid adding this variable by checking if the timer is running (i.e. the countdown variable is non-zero or `mChallengeInvalidTimer.IsRunning()`).